test/suite/test_checkpoint_snapshot05.py

```diff
@@ -140,38 +141,43 @@ def test_checkpoint_snapshot(self):
             else:
                 cursor1.set_value(self.valueb + str(i))
             self.assertEqual(cursor1.update(), 0)
-
-        # Create a checkpoint thread
+        
+        # Commit the transaction during a checkpoint.
         done = threading.Event()
         ckpt = checkpoint_thread(self.conn, done, checkpoint_count_max=1)
         try:
             ckpt.start()
             
-            # Wait for checkpoint to start and acquire its snapshot before committing.
+            # Wait for checkpoint to acquire its snapshot executing the commit.
             ckpt_snapshot = 0
             while not ckpt_snapshot:
-                time.sleep(1)
-                stat_cursor = self.session.open_cursor('statistics:', None, None)
-                ckpt_snapshot = stat_cursor[stat.conn.checkpoint_snapshot_acquired][2]
-                stat_cursor.close()
+                with open_cursor(self.session, 'statistics:') as stat_cursor:
+                    ckpt_snapshot = stat_cursor[stat.conn.checkpoint_snapshot_acquired][2]
+                time.sleep(0.5)
 
             session1.commit_transaction()
             self.evict(self.uri, ds, self.nrows)
         finally:
             done.set()
             ckpt.join()
 
-        #Take a backup and restore it.
+        # Verify exactly one checkpoint is present: as required by the test assertions.
+        # Assertion is NOT about the code under test, so use 'assert'.
+        with open_cursor(self.session, 'statistics:') as stat_cursor:
+            started = stat_cursor[stat.conn.checkpoints][2]
+            skipped = stat_cursor[stat.conn.checkpoint_skipped][2]
+        assert (started - skipped) == 1
+
+        # Take a backup and restore it.
         self.take_full_backup(".", self.backup_dir)
         self.reopen_conn(self.backup_dir)
 
         # Check the table contains the last checkpointed value.
         self.check(self.valuea, self.uri, self.nrows)
 
-        stat_cursor = self.session.open_cursor('statistics:', None, None)
-        inconsistent_ckpt = stat_cursor[stat.conn.txn_rts_inconsistent_ckpt][2]
-        keys_removed = stat_cursor[stat.conn.txn_rts_keys_removed][2]
-        stat_cursor.close()
+        with open_cursor(self.session, 'statistics:') as stat_cursor:
+            inconsistent_ckpt = stat_cursor[stat.conn.txn_rts_inconsistent_ckpt][2]
```

Comment:
Can we reduce the number of lines by doing the following?
```suggestion
            assert stat_cursor[stat.conn.txn_rts_inconsistent_ckpt][2] > 0
            assert stat_cursor[stat.conn.txn_rts_keys_removed][2] == 0
```

Reply:
I'm unsure. @kommiharibabu what do you think?

Reply:
I don't see why it would be an issue but I could be missing something.
I was just trying to find a way to remove the assignment to a variable and then the two following checks:
```
        self.assertGreater(inconsistent_ckpt, 0)
        self.assertEqual(keys_removed, 0)
```

Reply:
Yes, if we are going with the above approach, there is no need of the next two checks.